File: VB/E1984/TreeListFormatConditions.vb
Imports DevExpress.Utils.Serializing
Imports DevExpress.XtraTreeList.StyleFormatConditions

Namespace E1984
	Public Class TreeListFormatConditions

		Private _conditions As SerializableFormatCollection

		Public Sub New()
			_conditions = New SerializableFormatCollection()
		End Sub

		Public Sub New(ByVal styles As StyleFormatConditionCollection)
			_conditions = New SerializableFormatCollection(styles)
		End Sub

		<XtraSerializableProperty(XtraSerializationVisibility.Collection, True, False, True, 0, XtraSerializationFlags.None)>
		Public Conditions As Function(SerializableFormatCollection) _conditions

		Private Function XtraCreateConditionsItem(ByVal e As XtraItemEventArgs) As Object
			Return Conditions.Add()
		End Function
	End Class
End Namespace

File: CS/E1984/TreeListFormatConditions.cs
using DevExpress.Utils.Serializing;
using DevExpress.XtraTreeList.StyleFormatConditions;

namespace E1984 {
    public class TreeListFormatConditions
    {

        private SerializableFormatCollection _conditions;

        public TreeListFormatConditions()
        {
            _conditions = new SerializableFormatCollection();
        }

        public TreeListFormatConditions(StyleFormatConditionCollection styles)
        {
            _conditions = new SerializableFormatCollection(styles);
        }

        [XtraSerializableProperty(XtraSerializationVisibility.Collection, true, false, true, 0, XtraSerializationFlags.None)]
        public SerializableFormatCollection Conditions => _conditions;

        private object XtraCreateConditionsItem(XtraItemEventArgs e)
        {
            return Conditions.Add();
        }
    }
}
